### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by UncleJey** [simeonradivoev/UniGit#8]
Hi!
Great plugin!
Did it working for linux version? 
Ubuntu 16.10
Unity3d 5.2

have error
DllNotFoundException: git2-785d8c4
LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor ()
LibGit2Sharp.Core.NativeMethods..cctor ()
Rethrow as TypeInitializationException: An exception was thrown by the type initializer for LibGit2Sharp.Core.NativeMethods
LibGit2Sharp.Core.Handles.SafeHandleBase..ctor ()
LibGit2Sharp.Core.Handles.NullRepositorySafeHandle..ctor ()
LibGit2Sharp.Core.Proxy.git_repository_open_ext (System.String path, RepositoryOpenFlags flags, System.String ceilingDirs)
LibGit2Sharp.Repository.IsValid (System.String path)
UniGit.GitManager.get_IsValidRepo () (at D:/My Documents/My Pictures/My Art/Games/UniGit/Assets/Plugins/UniGit/Editor/GitManager.cs:588)
UniGit.GitManager.Initlize () (at D:/My Documents/My Pictures/My Art/Games/UniGit/Assets/Plugins/UniGit/Editor/GitManager.cs:74)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)

**Comment by simeonradivoev** [comment]
hm strange the error is coming from LibGit2Sharp. Specifically the libgit plugin. Can you make sure the plugin .SO file located in "Plugins/LibGit2Sharp/Editor/NativeBinaries/linux/amd64/libgit2-785d8c4.so". Maybe change its platform to work on all editors and see if you still have the same issue.

**Comment by UncleJey** [comment]
i thing settings is ok
https://gyazo.com/aae873ed902df8b1c9b7940972a86761

**Comment by simeonradivoev** [comment]
Try and turn the settings to any platform without excluded platforms.

**Comment by rolandoam** [comment]
had the same issue. Fixed by replacing `libgit2-785d8c4.so` with the one in my system (just `cp /usr/lib/libgit2.so libgit2-785d8c4.so`). It  was not loading because libgit2-nnn provided needed libcurl-ssl3 and my system had libcurl-ssl4. However, it still doesn't work for me, it says "Not a git repo" and the project clearly is a git repo (currently maintaining it from the cli).

**Comment by rolandoam** [comment]
ok... the problem is that it's expecting the full path to the project directory to be in (for some reason) a windows format (with backslashes). It actually creates a directory named "\home\username\path\to\project" where "/home/username/path/to/project" is the actual path in the root of the project and inside that directory it's where this plugin is expecting the git repo.

**Comment by simeonradivoev** [comment]
So I will have to account for slashes in linux ? I thought C# accounted for that.

**Comment by rolandoam** [comment]
I don't think you should worry about the path separator. It should be handled by C#. I guess it's how you're building the path. I'll try to take a look at your code to see if I can find the problem.

**Comment by rolandoam** [comment]
It might have to do with things like this:

https://github.com/simeonradivoev/UniGit/blob/580fab85f7830679ebde93eadd8dab4bbf328def/Assets/Plugins/UniGit/Editor/UniGitLoader.cs#L31

I'll try to give it a shot some time today.

**Comment by simeonradivoev** [comment]
Unity uses the linux '/' slash for its Asset Database and I had problems with gitlib2sharp on Windows, that's why I covert slashes from paths provided by unity
